set.mm

```diff
@@ -515900,6 +516454,485 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
   $}
 
 
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Subrings generated by a set
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  ${
+    elrgspn.b $e |- B = ( Base ` R ) $.
+    elrgspn.m $e |- M = ( mulGrp ` R ) $.
+    elrgspn.x $e |- .x. = ( .g ` R ) $.
+    elrgspn.n $e |- N = ( RingSpan ` R ) $.
+    elrgspn.f $e |- F = { f e. ( ZZ ^m Word A ) | f finSupp 0 } $.
+    elrgspn.r $e |- ( ph -> R e. Ring ) $.
+    elrgspn.a $e |- ( ph -> A C_ B ) $.
+    ${
+      $d .x. a b f g i w $.  $d .x. a f g i v w $.  $d .x. a g i u v w $.
```

Comment:
What an amazing set of $d conditions. Do you think it could be optimized? For example, I see `$d L u $.` and then `$d L a f g i u v w $.`, so the first one seems redundant. Maybe we have some tooling (mmj2?) which can remove redundant disjoint conditions?

Reply:
Yes, I'm pretty sure it can be optimized. I just found out there is a `check-dvs` tool which could help.

Reply:
I wrote a program to optimize $d conditions using a SAT solver and a clique edge covering problem, this is the smallest set of $d conditions possible (8 in total):

`$d a e f g h i j ph s t u v w x y z $. $d R a e f g h i j s t u v w x y $. $d M a e f g h i j s u v w x y $. $d .x. a e f g h i j s u v w x y $. $d F a e f g h i j s t u v w x y z $. $d S g i s t w x y z $. $d A a e f g h i j s t u v w x y z $. $d B a f g j t u v w y $.`

It doesn't add any new disjoint conditions between variables, of course.